src/measure_theory/integral/set_integral.lean

```diff
@@ -787,6 +787,105 @@ end
 
 end tendsto_mono
 
+/-! ### First moment method -/
+
+section first_moment
+variables {μ : measure α} {s N : set α} {f : α → ℝ} {r : ℝ}
+
+/-- **First moment method**. An integrable function is smaller than its mean on a set of positive
+measure. -/
+lemma measure_le_set_integral_pos (hμ : μ s ≠ 0) (hμ₁ : μ s ≠ ∞) (hf : integrable_on f s μ)
+  (hs : null_measurable_set s μ) :
+  0 < μ {x ∈ s | f x ≤ ∫ a in s, f a ∂μ / (μ s).to_real} :=
+begin
+  obtain ⟨t, hts, ht, hμts⟩ := hs.exists_measurable_subset_ae_eq,
+  replace hf := hf.mono_set hts,
+  simp_rw [←set_of_inter_eq_sep, ←set_integral_congr_set_ae hμts, ←measure_congr hμts,
+    ←measure_congr ((eventually_eq.refl _ _).inter hμts)],
+  rw ←measure_congr hμts at hμ hμ₁,
+  haveI : fact (μ t < ∞) := ⟨lt_top_iff_ne_top.2 hμ₁⟩,
+  refine pos_iff_ne_zero.2 (λ H, _),
+  have : 0 < μ (support (f - const _ (∫ a in t, f a ∂μ / (μ t).to_real)) ∩ t),
+  { rwa [pos_iff_ne_zero, inter_comm, ←diff_compl, ←diff_inter_self_eq_diff, measure_diff_null],
+    exact eq_bot_mono (measure_mono $ inter_subset_inter_left _ $ λ a ha,
+      (sub_eq_zero.1 $ of_not_not ha).le) H },
+  rw [←set_integral_pos_iff_support_of_nonneg_ae _ (hf.sub $ integrable_const _)] at this,
+  replace hμ := (ennreal.to_real_pos hμ hμ₁).ne',
+  simpa [hμ, integral_sub hf, set_integral_const, smul_eq_mul, mul_div_cancel'] using this,
+  change _ = _,
+  simp only [compl_set_of, ht, pi.zero_apply, pi.sub_apply, sub_nonneg, not_le,
+    measure.restrict_apply'],
+  refine eq_bot_mono (measure_mono $ inter_subset_inter_left _ $ λ a ha, _) H,
+  dsimp at ha,
+  exact ha.le,
+end
+
+/-- **First moment method**. An integrable function is greater than its mean on a set of positive
+measure. -/
+lemma measure_set_integral_le_pos (hμ : μ s ≠ 0) (hμ₁ : μ s ≠ ∞) (hf : integrable_on f s μ)
+  (hs : null_measurable_set s μ) :
+  0 < μ {x ∈ s | ∫ a in s, f a ∂μ / (μ s).to_real ≤ f x} :=
+by simpa [integral_neg, neg_div] using measure_le_set_integral_pos hμ hμ₁ hf.neg hs
+
+/-- **First moment method**. The minimum of an integrable function is smaller than its mean. -/
+lemma exists_set_le_integral (hμ : μ s ≠ 0) (hμ₁ : μ s ≠ ∞) (hf : integrable_on f s μ)
+  (hs : null_measurable_set s μ) :
+  ∃ x ∈ s, f x ≤ ∫ a in s, f a ∂μ / (μ s).to_real :=
+let ⟨x, hx, h⟩ := nonempty_of_measure_ne_zero (measure_le_set_integral_pos hμ hμ₁ hf hs).ne'
+  in ⟨x, hx, h⟩
+
+/-- **First moment method**. The maximum of an integrable function is greater than its mean. -/
+lemma exists_set_integral_le (hμ : μ s ≠ 0) (hμ₁ : μ s ≠ ∞) (hf : integrable_on f s μ)
+  (hs : null_measurable_set s μ) :
+  ∃ x ∈ s, ∫ a in s, f a ∂μ / (μ s).to_real ≤ f x :=
+let ⟨x, hx, h⟩ := nonempty_of_measure_ne_zero (measure_set_integral_le_pos hμ hμ₁ hf hs).ne'
+  in ⟨x, hx, h⟩
+
+variables [is_finite_measure μ]
+
+/-- **First moment method**. An integrable function is smaller than its mean on a set of positive
+measure. -/
+lemma measure_le_integral_pos (hμ : μ ≠ 0) (hf : integrable f μ) :
```

Comment:
Could you also give versions (of this lemma and the next ones) assuming that mu is a probability meaure, removing the assumption `hμ` and also `(μ univ).to_real` from the conclusion?

Reply:
This suggestion is less useful now that everything is stated in terms of `average` and we have `average_eq_integral`, but I have still done it.